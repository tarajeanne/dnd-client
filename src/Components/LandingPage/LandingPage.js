import React from 'react';
import { Redirect } from 'react-router-dom';

class LandingPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      value: ''
    };
  }

  handleChange = (event) => {
    this.setState({value: event.target.value});
  }

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.updateCharacter(this.state.value);
    this.props.changeScreen('class')
    this.setState({
      toRace: true
    });
  }

  render() {
    if (this.state.toRace) {
      return <Redirect to={'/race'} />
    }
    return (
      <section>
        <header>Welcome to the Dungeons and Dragons character creator for true beginners!</header>
        <form onSubmit={this.handleSubmit}>
          <label for="name-input">Please enter a name for your character:</label>
          <input 
            name="name-input" 
            value={this.state.value} 
            onChange= {this.handleChange}
            id="name-input" 
            type="text" 
            placeholder="Sabrill of Savernke">
          </input>
        </form>
      </section>
      

    )
  }
}

export default LandingPage;