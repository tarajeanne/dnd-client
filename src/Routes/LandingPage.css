.landing-page {
  display: flex;
  flex-flow: row-reverse wrap;
  justify-content: space-between;
  max-width: 1200px;
  margin: auto;
  padding: 20px 10px;
}

.login-toggle {
  text-align: center;
}

.landing-page-login {
  display: inline-block;
  height: 100%;
  margin: auto;
}

.landing-page-info {
  display: inline-block;
  margin: auto;
}

.video-container {
  display: inline-block;
  position: relative;
  margin: auto;
  width: 100%;
  height: 0;
  padding-bottom: 51%;
}

.video {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
}

@media screen and (min-width: 750px) {
  .landing-page {
    flex-flow: row-reverse nowrap;
  }

  .LoginPage {
    width: 300px;
  }

}
