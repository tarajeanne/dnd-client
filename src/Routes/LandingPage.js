import React, { Component } from 'react'
import LoginForm from '../Components/LoginForm/LoginForm';
import { Link } from 'react-router-dom';

export default class LoginPage extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  }

  handleLoginSuccess = () => {
    const { location, history } = this.props
    const destination = (location.state || {}).from || '/user'
    history.push(destination)
  }

  render() {
    return (
      <div className='LoginPage'>
        <h2>Login</h2>
        <LoginForm
          onLoginSuccess={this.handleLoginSuccess}
        />
        <p>
          If you just want to try out the app, log in under the credentials username: testuser and password: testuser. Characters under this account will be deleted occassionally. 
        </p>
        <h2>
          No account? Register:
        </h2>
        <Link to="/register">Register</Link>
      </div>
    )
  }
}
