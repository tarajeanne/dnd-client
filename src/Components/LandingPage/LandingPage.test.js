import React from 'react';
import LandingPage from './LandingPage';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('LandingPage', () => {
    it('renders as expected', () => {
        const wrapper = shallow(<LandingPage />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
  });
