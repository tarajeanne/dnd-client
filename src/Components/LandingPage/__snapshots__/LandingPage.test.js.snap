// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage renders as expected 1`] = `
<section>
  <header>
    Welcome to the Dungeons and Dragons character creator for true beginners!
  </header>
  <form
    onSubmit={[Function]}
  >
    <label
      for="name-input"
    >
      Please enter a name for your character:
    </label>
    <input
      id="name-input"
      name="name-input"
      onChange={[Function]}
      placeholder="Sabrill of Savernke"
      type="text"
      value=""
    />
  </form>
</section>
`;
